components/org.wso2.micro.integrator.core/src/main/java/org/wso2/micro/core/CarbonAxisConfigurator.java

```diff
@@ -385,30 +387,26 @@ public void setRepoUpdateFailed() {
 
     @Override
     protected void startSearch(RepositoryListener listener) {
-        //TODO need to implement
- /*       schedulerTask = new CarbonDeploymentSchedulerTask(listener,
-                                                          axisConfig,
-                                                          MultitenantConstants.SUPER_TENANT_ID,
-                                                          MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
-        scheduler = Executors
-                .newScheduledThreadPool(1, new CarbonThreadFactory(new ThreadGroup("CarbonDeploymentSchedulerThread")));
-        String deploymentInterval =
-                CoreServerInitializer.
-                        getServerConfigurationService().getFirstProperty("Axis2Config.DeploymentUpdateInterval");
+        CarbonDeploymentSchedulerTask schedulerTask = new CarbonDeploymentSchedulerTask(listener, axisConfig);
+        scheduler = Executors.newScheduledThreadPool(1, new CarbonThreadFactory(
+                new ThreadGroup("CarbonDeploymentSchedulerThread")));
+
+        String deploymentInterval = CarbonCoreDataHolder.getInstance().getServerConfigurationService().getFirstProperty(
+                "Axis2Config.DeploymentUpdateInterval");
         int deploymentIntervalInt = 15;
-        if(deploymentInterval != null) {
-           deploymentIntervalInt = Integer.parseInt(deploymentInterval);
+        if (deploymentInterval != null) {
+            deploymentIntervalInt = Integer.parseInt(deploymentInterval);
```

Comment:
We need to consider parse exception here ?

Reply:
Done.